platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/gossip/modular/SyncGossipModular.java

```diff
@@ -102,10 +111,26 @@ public SyncGossipModular(
             @NonNull final IntakeEventCounter intakeEventCounter,
             @NonNull final PlatformStateFacade platformStateFacade) {
 
-        this.network = new PeerCommunication(platformContext, roster, selfId, keysAndCerts);
+        final RosterEntry selfEntry = RosterUtils.getRosterEntry(roster, selfId.id());
+        final X509Certificate selfCert = RosterUtils.fetchGossipCaCertificate(selfEntry);
+        final List<PeerInfo> peers;
+        if (!CryptoStatic.checkCertificate(selfCert)) {
+            // Do not make peer connections if the self node does not have a valid signing certificate in the roster.
+            // https://github.com/hashgraph/hedera-services/issues/16648
+            logger.error(
+                    EXCEPTION.getMarker(),
+                    "The gossip certificate for node {} is missing or invalid. "
+                            + "This node will not connect to any peers.",
+                    selfId);
+            peers = Collections.emptyList();
+        } else {
+            peers = Utilities.createPeerInfoList(roster, selfId);
+        }
+        var selfPeer = Utilities.toPeerInfo(selfEntry);
```

Comment:
```suggestion
        final PeerInfo selfPeer = Utilities.toPeerInfo(selfEntry);
```
Please replace `var` declarations with explicit types and add `final` modifiers where possible. This aligns with our platform's code style. I get that there was other usages of `var` before you modified the file please update them as well.